STUDIO/org.openl.rules.repository.azure/src/org/openl/rules/repository/azure/AzureBlobRepository.java

```diff
@@ -0,0 +1,760 @@
+package org.openl.rules.repository.azure;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.azure.storage.blob.models.*;
+import com.azure.storage.blob.options.BlobParallelUploadOptions;
+import com.azure.storage.common.implementation.Constants;
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.openl.rules.repository.api.*;
+import org.openl.rules.repository.common.ChangesMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.TypeDescription;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.representer.Representer;
+
+public class AzureBlobRepository implements FolderRepository {
+    private static final String UNSUPPORTED_IN_FOLDER_REPOSITORY = "Unsupported in folder repository";
+    private static final String VERSION_FILE = "versions.yaml";
+
+    private static final String MODIFICATION_FILE = "[openl]/.modification";
+    private static final String CONTENT_PREFIX = "[content]/";
+    private static final String VERSIONS_PREFIX = "[openl]/versions/";
+
+    private final Logger log = LoggerFactory.getLogger(AzureBlobRepository.class);
+
+    private String id;
+    private String name;
+    private ChangesMonitor monitor;
+    private int listenerTimerPeriod = 10;
+    private String uri;
+
+    private BlobContainerClient blobContainerClient;
+    private PassiveExpiringMap<CacheKey, AzureCommit> commitsCache;
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setMonitor(ChangesMonitor monitor) {
+        this.monitor = monitor;
+    }
+
+    public void setListenerTimerPeriod(int listenerTimerPeriod) {
+        this.listenerTimerPeriod = listenerTimerPeriod;
+    }
+
+    public void setUri(String uri) {
+        this.uri = uri;
+    }
+
+    public void initialize() {
+        blobContainerClient = new BlobContainerClientBuilder()
+                .endpoint(uri)
+                .buildClient();

```

Comment:
please define ability to use credentials for building azure connection, like:
```
if (StringUtils.isNotBlank(username)) {
    builder.credential(new BasicAuthenticationCredential(username, password));
}
```
and/or ``StorageSharedKeyCredential``

Reply:
Do we really need it? Different URL can be generated for different users with different access rights. Isn't it enough?

Reply:
I think about using Azure Emulator like https://hub.docker.com/_/microsoft-azure-storage-azurite . Does current implementation support it? For development purpose using of storage emulator is more likely rather than real one

Reply:
@sanail Which authentication are supported by this repository? If a token in the URL is enough, then no needs to introduce login/password/tocken settings.
https://docs.microsoft.com/en-us/azure/storage/common/authorize-data-access?toc=/azure/storage/blobs/toc.json

Reply:
@yurkom For now Shared access signature (SAS) (it's expected that it will be URL part) is a primary authentication method.

Also added support for Shared Key (storage account key) but it can't be configured through UI, only from our property files.

Login/passord isn't supported.